custom/sensors/PDP.java

```diff
@@ -75,38 +126,89 @@ public double getVoltage() {
 	 *         If PDP connection is lost, InvalidSensorException will be thrown.
 	 */
 	public double getVoltageSafely() throws InvalidSensorException {
-		byte[] rawArray = status3.read();
-		if (rawArray != null) {
-			int rawVoltage = rawArray[6] & 0xFF; // busVoltage is the 7th byte of the third status
-			cachedVoltage = (rawVoltage) * 0.05 + 4.0;
-			lastRead = System.currentTimeMillis();
-		}
+		readStatus3();
 		if (System.currentTimeMillis() - lastRead > PDP.MAX_AGE) {
```

Comment:
Since this code is duplicated, maybe break it out into its own function or implement it inside the `readStatus` methods?